documentation/reference/ci_cd.md

```diff
@@ -0,0 +1,491 @@
+  do_publish: "true" | "false" // whether one or more images are to be published
+  publish_matrix: {
+    image: ("catalog" | "ingestion_server" | "api" | "api_nginx" | "frontend")[] // the name of the image to be published
+  }
+}
+```
+
+## Universal jobs
+
+### `build-images`
+
+Builds images as dictated by the [`determine-images`](#determine-images) job.
+Since the images are built in a matrix, some image-specific steps are
+conditionally run using an `if` expression. Since any unused Docker build
+arguments have no effect, the job simply populates all build arguments needed by
+all images.
+
+Images built by this job are published as `.tar` artifacts and can be loaded
+into other jobs that need them using the [`load-img`](#load-img) action.
+
+This job is only run if there is at least one image needed to build, based on
+the `do_build` output of the [`determine-images`](#determine-images) job.
+
+## Catalog jobs
+
+### `test-cat`
+
+Runs tests for the catalog using the `catalog/test` recipe. Tests are run inside
+a Docker container so neither Python nor Node.js needs to be installed.
+
+This job is skipped if the catalog codebase has not changed. Its counterparts
+are [`test-ing`](#test-ing) for the ingestion server, [`test-api`](#test-api)
+for the API and [`nuxt-build`](#nuxt-build) for the frontend.
+
+```{note}
+This job is treated as the proof of functionality for publishing Docker images
+for the catalog.
+```
+
+### `catalog-checks`
+
+Runs tests for the catalog using the `catalog/generate-dag-docs` recipe. The job
+generates a new file from a Docker container so Python does not need to be
+installed. However, the job runs linting on the new file, so Node.js needs to be
+set up.
+
+This job is skipped if the catalog codebase has not changed.
+
+## Ingestion server jobs
+
+### `test-ing`
+
+Runs tests for the ingestion server using the `ingestion_server/test-local`
+recipe. Tests are run on the host so Python needs to be installed and set up.
+
+This job is skipped if the ingestion server codebase has not changed. Its
+counterparts are [`test-cat`](#test-cat) for the catalog,
+[`test-api`](#test-api) for the API and [`nuxt-build`](#nuxt-build) for the
+frontend.
+
+```{note}
+This job is treated as the proof of functionality for publishing Docker images
+for the ingestion server.
+```
+
+## API jobs
+
+### `test-api`
+
+Initialises the API using the `api/init` recipe and runs tests for the API using
+the `api/test` recipe. Tests are run inside a Docker container so neither Python
+nor Node.js needs to be installed.
+
+This job is skipped if the API codebase has not changed. Its counterparts are
+[`test-cat`](#test-cat) for the catalog, [`test-ing`](#test-ing) for the
+ingestion server and [`nuxt-build`](#nuxt-build) for the frontend.
+
+```{note}
+This job is treated as the proof of functionality for publishing Docker images
+for the API. Since it also initialises the API, it also largely proves that the
+ingestion server is working.
```

Comment:
```suggestion
for the API. Since API tests depend on the initialisation of sample data handled by the ingestion server, this job also provides a basic verification of core ingestion server functionality.
```

I'd shy away from saying that running the init script "largely" proves the ingestion server is working. The init script does not (intentionally or explicitly) exercise every aspect of even the common route of the ingestion server like cleanup.

Reply:
Updated in a87e889.